logproof/src/bfv_statement.rs

```diff
@@ -0,0 +1,863 @@
+//! This module provides a mid-level API for generating SDLP prover and verifier knowledge from BFV
+//! encryptions, all at the [`seal_fhe`] layer.
+
+use std::{fmt::Debug, marker::PhantomData, ops::Neg};
+
+use crypto_bigint::{NonZero, Uint};
+use seal_fhe::{
+    Ciphertext, Context, EncryptionParameters, Plaintext, PolynomialArray, PublicKey, SecretKey,
+};
+use sunscreen_math::{
+    poly::Polynomial,
+    ring::{BarrettBackend, BarrettConfig, Ring, Zq},
+    One, Zero,
+};
+
+use crate::{
+    linear_algebra::{Matrix, PolynomialMatrix},
+    rings::ZqRistretto,
+    Bounds, LogProofProverKnowledge, LogProofVerifierKnowledge,
+};
+
+/// In SEAL, `u` is sampled from a ternary distribution.
+// TODO if bounds are in u64 then perhaps this should be 3?
+const U_COEFFICIENT_BOUND: u64 = 2;
```

Comment:
Calling this out for those more familiar... seems like it should be 3? But existing bounds were using 2 on the ternary polynomials.

Reply:
The 2 just refers to the max absolute value of the bound. So since the value can be in {-1, 0, 1}, the bound should be 2. I believe, correct me if I am remembering incorrectly.

Reply:
I think you are right, after looking [here](https://github.com/Sunscreen-tech/Sunscreen/blob/5cccd88ac0ee4ab53287a4ee0a0fed2126abd2b4/logproof/src/linear_relation.rs#L1138). I've removed the TODO comments.

Also the tests would have undoubtedly failed by now if the bound 2 was too low, so there's that.